pkg/hive/hive.go

```diff
@@ -0,0 +1,223 @@
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
I assume this file will "never" be modified by devs, right?

Reply:
Yeah I'd expect this not to change much after this initial iteration. Why do you ask?

Reply:
out of curiosity, that is all :)